src/query_ctx.c

```diff
@@ -126,39 +142,42 @@ static void _QueryCtx_ThreadSafeContextUnlock(QueryCtx *ctx) {
 
 bool QueryCtx_LockForCommit(void) {
 	QueryCtx *ctx = _QueryCtx_GetCtx();
-	// Lock GIL.
-	RedisModuleCtx *redis_ctx = ctx->global_exec_ctx.redis_ctx;
-	GraphContext *gc = ctx->gc;
-	RedisModuleString *graphID = RedisModule_CreateString(redis_ctx, gc->graph_name,
-														  strlen(gc->graph_name));
-	char *error;
-	_QueryCtx_ThreadSafeContextLock(ctx);
-	// Open key and verify.
-	RedisModuleKey *key = RedisModule_OpenKey(redis_ctx, graphID, REDISMODULE_WRITE);
-	RedisModule_FreeString(redis_ctx, graphID);
-	if(RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) {
-		asprintf(&error, "Encountered an empty key when opened key %s",
-				 ctx->gc->graph_name);
-		QueryCtx_SetError(error);
-
-		goto clean_up;
+	if(!ctx->internal_exec_ctx.locked_for_commit) {
```

Comment:
I think it's more readable to return early:
`if(ctx->internal_exec_ctx.locked_for_commit) return true;`

But they're functionally identical, no big deal.

Reply:
done